### PR TITLE
Init firebase

### PR DESCRIPTION
**Pull request by aaronfinsrud**
## Description:
Implemented basic authentication functionality (login, logout, signup, private routes)

## Context:
These changes allow users to Login, Sign up and Log out. Private Routes can be easily created as soon as development is complete.

## Checklist:
[x] Implement AuthProvider context wrapper component
[x] Configure Firebase & Firestore
[x] Integrate SignUp, and LogIn component w/ Firebase auth
[x] Implement PrivateRoutes wrapper component

## Notes:
Remember to run `npm install` to install any newly added dependencies
And remember to add firebase config variables to the config.js

### Screenshots (if any):

### CODE REVIEW

File: client/src/contexts/index.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  function updateEmail(newEmail) {
+    return updateemail(currentUser, newEmail);
+  }
+
+  function updatePassword(newPassword) {
+    return updatepassword(currentUser, newPassword);
+  }
+  // {displayName: 'displayName', photoURL: 'urlHere'}
```
Comment (alexander-huerta) [inline client/src/contexts/index.jsx:46]: do you want to mereg with this comment?

Comment (alexander-huerta) [inline client/src/contexts/index.jsx:46]: do you want to mereg with this comment?

File: client/src/components/Login.jsx [hunk 2]
```diff
@@ -1,16 +1,68 @@
...
+
+  // Methods
+  const onChange = (e) => {
+    const { id, value } = e.target;
+    if (id === 'email') setEmail(value);
+    if (id === 'password') setPassword(value);
+  };
+
+  const handleSubmit = async (event) => {
     event.preventDefault();
```
Comment (alexander-huerta) [inline client/src/components/Login.jsx]: on line 25 the event was referenced with 'e' and on line 31 its referenced with 'event'. I think we should try to pick one format and stick with it

File: client/src/contexts/index.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (user) => {
+      setCurrentUser(user);
+      setLoading(false);
+    });
+
+    return unsubscribe;
+  }, []);
+
+  const value = useMemo(() => ({
```
Comment (alexander-huerta) [inline client/src/contexts/index.jsx:60]: very useful hook! nice

**Review by alexander-huerta — APPROVED** [review]
Nice job on the log in and authentication. Overall easy to follow and well organized. It will be that much easier to add additional functionality to the codebase when the time comes.

File: client/src/components/Login.jsx (hunk 2 above)
Comment (aaronfinsrud) [reply client/src/components/Login.jsx]: Good point. I'll update that

File: client/src/contexts/index.jsx (hunk 1 above)
Comment (aaronfinsrud) [reply client/src/contexts/index.jsx:46]: Yeah -I'm going to come back to that in my next PR. Thanks for pointing it out

**Pull request merged**